crates/macro-support/src/parser.rs

```diff
@@ -19,29 +19,59 @@ use crate::ClassMarker;
 
 thread_local!(static ATTRS: AttributeParseState = Default::default());
 
-/// Javascript keywords which are not keywords in Rust.
-const JS_KEYWORDS: [&str; 20] = [
-    "class",
+/// Javascript keywords.
+///
+/// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#reserved_words
+const JS_KEYWORDS: [&str; 44] = [
+    "arguments", // reserved in strict mode, which we are in
+    "break",
     "case",
     "catch",
+    "class",
+    "const",
+    "continue",
     "debugger",
     "default",
     "delete",
+    "do",
+    "else",
+    "enum", // always reserved, but not used
+    // "eval", reserved in strict mode, but behaves like a functions in JS, so we allow it
     "export",
     "extends",
+    // "false", false is a keyword, but behaves like a literal in JS
```

Comment:
Same as `eval`. But needs yet another type of list.
Could be inline as well, how you implement this is at your discretion.